lib/statistics_to_messages.cpp

```diff
@@ -0,0 +1,88 @@
+// PredictHaplo-Paired: a program for estimating haplotypes from "next
+// generation sequencing" reads
+//
+// Copyright (C) 2022 ETH Zurich
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "phaplo/statistics_to_messages.hpp"
+#include "phaplo/ParseStatistics.hpp"
+#include <iomanip>
+#include <string>
+#include <vector>
+
+namespace phaplo {
+
+namespace {
+std::string to_percentage(const std::size_t value, const std::size_t total) {
+  std::stringstream stream;
+  stream << std::setprecision(1) << std::fixed;
+  stream << 100. * static_cast<double>(value) / static_cast<double>(total)
+         << "\%";
+  return stream.str();
+}
+} // namespace
+
+std::vector<std::string>
+statistics_to_messages(const ParseStatistics &statistics) noexcept {
+  auto r = std::vector<std::string>();
+
+  const auto to_reads_percentage = [&](const std::size_t x) {
+    return to_percentage(x, statistics.reads.total_count);
+  };
+  const auto to_pairs_percentage = [&](const std::size_t x) {
+    return to_percentage(x, statistics.pairs.total_count);
+  };
+
+  if (statistics.reads.unsupported_attribute) {
```

Comment:
Would it be possible to list here which used attribute was unsupported?

Reply:
Thanks for the suggestion! Since the attribute could be different for every read, I only see the option of splitting up the warning into individual warnings per attribute/bit so far. Is this what you have in mind?

Reply:
Yes, let's keep it like this for now.

Not sure how precise we can actually make these warnings. Wouldn't we always just list what was not understood?

Reply:
Perfect, thanks! Yes, that's possible; let's decide when we encounter a use case.